File: 20140405/fizzbuzz.py
def toStr(num):
    if (num%15==0):
        return "FizzBuzz"
    elif (num%3==0):
        return "Fizz"
    elif (num%5==0):
        return "Fizz"
    else:
        return num

def fizzBuzz(min, max):
    for i in range(min,max):
        print(toStr(i))

fizzBuzz(1,100)

File: 20140125/fizzbuzz.rb
class FizzBuzz
  def initialize(min, max)
    @min = min
    @max = max
  end

  def fizz? num
    return (num % 3) == 0
  end
  private :fizz?

  def buzz? num
    return (num % 5) == 0
  end
  private :buzz?

  def run
    for i in @min..@max
      if fizz? i and buzz? i
        p "FizzBuzz"
      elsif fizz? i
        p "Fizz"
      elsif buzz? i
        p "Buzz"
      else
        p i
      end
    end
  end
end

fb = FizzBuzz.new(1, 100)
fb.run
